opsdroid/__main__.py

```diff
@@ -110,6 +111,13 @@ def print_example_config(ctx, param, value):
     ctx.exit(0)
 
 
+def edit_config(ctx, param, value='open'):
+    """Easy way to edit the config file."""
+    subprocess.run([value, DEFAULT_CONFIG_PATH])
```

Comment:
I think `open` is a Mac specific command so this will not work on Linux or Windows.

There is an environment variable called `EDITOR` which is used by tools like `git` and `crontab` to know how to open a file for editing. 

As opsdroid is a command line application I think we should be using the command line editor functionality. So perhaps check `EDITOR` and if it hasn't been set then run `vi`. That way users can set sublime, vscode, atom or whatever to their default editor and they should get the same functionality as with `open`.